src/search.py

```diff
@@ -0,0 +1,148 @@
+from typing import Dict
+from typing import List
+
+import pandas as pd
+import yaml
+
+from src import logger
+
+
+def load_config(config_path: str) -> dict:
+    """Load yaml config using safe loader"""
+    with open(config_path, "r") as stream:
+        try:
+            config = yaml.safe_load(stream)
+            return config
+        except yaml.YAMLError as exc:
+            raise exc
+
+
+class SearchDataset:
+    """Search dataset using keyword and vector searches"""
+
+    def __init__(
+        self,
+        Dataset,  # noqa
+        data_df: pd.DataFrame,
+        config_path: str,
+    ) -> None:
+        """Initialise search object"""
+        self.Dataset = Dataset
+        self.data_df = data_df
+        self.config = load_config(config_path)
+        self._n_keyword_results = 100000
+        self._n_vector_results = 5000
+        self.keyword_matches_df = None
+        self.vector_matches_df = None
+        self.final_matches_df = None
+
+    @staticmethod
+    def create_OR_query(keywords: List[str]) -> str:
```

Comment:
Just so I understand - does this mean that all the multiword keywords such as "threat intelligence" will be run as "threat OR intelligence"? Ditto "cyber attack" = "cyber OR attack" etc?

Reply:
Thanks Rosie, no, it will look like this `"('threat intelligence') OR ('cyber attack')"`